### PR TITLE
refactor: enable errorlint and refactor code

### PR DESCRIPTION
**Pull request by hoanguyenkh**
**What type of PR is this?**

> Other

**What does this PR do? Why is it needed?**
- Enable errorlint
- Refactor code, 
- Remove directive errors comparison
- Make fmt.Errorf use %w to wrap error instead of %v

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

**Review by terencechain — DISMISSED** [review]

**Comment by hoanguyenkh** [comment]
hi @nalepae @saolyn can you merge PR?

File: beacon-chain/node/registration/p2p.go [hunk 1]
```diff
@@ -50,10 +50,9 @@ func readbootNodes(fileName string) ([]string, error) {
 	listNodes := make([]string, 0)
 	err = yaml.UnmarshalStrict(fileContent, &listNodes)
 	if err != nil {
-		if _, ok := err.(*yaml.TypeError); !ok {
+		var typeError *yaml.TypeError
+		if !errors.As(err, &typeError) {
 			return nil, err
-		} else {
-			log.WithError(err).Error("There were some issues parsing the bootnodes from a yaml file.")
 		}
```
Comment (prestonvanloon) [inline beacon-chain/node/registration/p2p.go:56]: This is a logic change. Please revert

File: beacon-chain/rpc/eth/beacon/handlers_validator.go [hunk 2]
```diff
@@ -8,6 +8,8 @@ import (
 	"strconv"
 	"strings"
 
+	"github.com/pkg/errors"
+
```
Comment (prestonvanloon) [inline beacon-chain/rpc/eth/beacon/handlers_validator.go:12]:
```suggestion
```
You may need to run gofmt or goimports too

File: beacon-chain/rpc/eth/shared/errors.go [hunk 3]
```diff
@@ -3,6 +3,8 @@ package shared
 import (
 	"net/http"
 
+	"github.com/pkg/errors"
+
```
Comment (prestonvanloon) [inline beacon-chain/rpc/eth/shared/errors.go:7]:
```suggestion
```

You may need to run gofmt or goimports too

File: beacon-chain/sync/rpc.go [hunk 4]
```diff
@@ -7,6 +7,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/pkg/errors"
+
```
Comment (prestonvanloon) [inline beacon-chain/sync/rpc.go:11]:
```suggestion
```
You may need to run gofmt or goimports too

File: beacon-chain/sync/rpc_beacon_blocks_by_range_test.go [hunk 5]
```diff
@@ -2,6 +2,7 @@ package sync
 
 import (
 	"context"
+	"github.com/pkg/errors"
```
Comment (prestonvanloon) [inline beacon-chain/sync/rpc_beacon_blocks_by_range_test.go:5]: Please put this import with the other non-standard imports. The second section starting on line 12

File: cmd/prysmctl/p2p/handler.go [hunk 6]
```diff
@@ -6,6 +6,8 @@ import (
 	"runtime/debug"
 	"strings"
 
+	"github.com/pkg/errors"
+
```
Comment (prestonvanloon) [inline cmd/prysmctl/p2p/handler.go:10]:
```suggestion
```

You may need to run gofmt or goimports too

File: config/params/loader.go [hunk 7]
```diff
@@ -53,10 +53,9 @@ func UnmarshalConfig(yamlFile []byte, conf *BeaconChainConfig) (*BeaconChainConf
 	}
 	yamlFile = []byte(strings.Join(lines, "\n"))
 	if err := yaml.UnmarshalStrict(yamlFile, conf); err != nil {
-		if _, ok := err.(*yaml.TypeError); !ok {
+		var typeError *yaml.TypeError
+		if !errors.As(err, &typeError) {
 			return nil, errors.Wrap(err, "Failed to parse chain config yaml file.")
-		} else {
-			log.WithError(err).Error("There were some issues parsing the config from a yaml file")
 		}
```
Comment (prestonvanloon) [inline config/params/loader.go:59]: This is a logic change. Please revert

File: tools/replay-http/main.go [hunk 8]
```diff
@@ -16,6 +16,8 @@ import (
 	"os"
 	"path"
 
+	"github.com/pkg/errors"
+
```
Comment (prestonvanloon) [inline tools/replay-http/main.go:20]:
```suggestion
```

You may need to run gofmt or goimports too

File: validator/graffiti/parse_graffiti.go [hunk 9]
```diff
@@ -32,10 +34,9 @@ func ParseGraffitiFile(f string) (*Graffiti, error) {
...
 	g := &Graffiti{}
 	if err := yaml.UnmarshalStrict(yamlFile, g); err != nil {
-		if _, ok := err.(*yaml.TypeError); !ok {
+		var typeError *yaml.TypeError
+		if !errors.As(err, &typeError) {
 			return nil, err
-		} else {
-			log.WithError(err).Error("There were some issues parsing graffiti from a yaml file.")
 		}
 	}
```
Comment (prestonvanloon) [inline validator/graffiti/parse_graffiti.go:41]: This is a logic change, please revert

**Review by prestonvanloon — CHANGES_REQUESTED** [review]
Apologies for the eager approval. Upon closer review, there are some things that need to be addressed. Thanks

**Comment by prestonvanloon** [comment]
You need to run gazelle on this PR.

```
bazel run //:gazelle
```

**Comment by hoanguyenkh** [comment]
thank @prestonvanloon for feedback, please check it again! thank you

**Review by prestonvanloon — APPROVED** [review]
LGTM, thank you!!

**Comment by hoanguyenkh** [comment]
can you help me merge it @prestonvanloon

**Pull request merged**